File: PerfTest2/PerfTest2-Java/app/src/main/java/com/vandammeford/kevinf/perftest2_java/SqLiteTableAdapter.java
package com.vandammeford.kevinf.perftest2_java;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.TextView;

import java.util.ArrayList;

public class SqLiteTableAdapter extends BaseAdapter {
    private Context context;
    private ArrayList<String> records;
    private SqLiteDisplayType displayType;

    public SqLiteTableAdapter(Context context, SqLiteDisplayType displayType)
    {
        this.context = context;
        this.displayType = displayType;
        loadRecords();
    }

    private void loadRecords() {
        SqLiteUtilities utilities;

        utilities = new SqLiteUtilities(this.context);
        try {
            utilities.openConnection();
            if (displayType == SqLiteDisplayType.ShowAll) {
                records = utilities.getAllRecords();
            } else {
                records = utilities.getRecordsWith1();
            }
            utilities.closeConnection();
        } catch (Exception ex) {

        }
    }

    @Override
    public int getCount() {
        return records.size();
    }

    @Override
    public Object getItem(int item) {
        // TODO Auto-generated method stub
        return records.get(item);
    }

    @Override
    public long getItemId(int position) {
        // TODO Auto-generated method stub
        return position;
    }

    @Override
    public View getView(int position, View convertView, ViewGroup viewGroup) {

        if(convertView == null)
        {
            LayoutInflater inflater = (LayoutInflater)context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
            convertView = inflater.inflate(android.R.layout.simple_list_item_1, null);
        }

        TextView txtItem = (TextView)convertView.findViewById(android.R.id.text1);
        txtItem.setText(records.get(position));

        return convertView;
    }
}

File: PerfTest2/PerfTest2-Xamarin/PerfTest2-ClassicXamarin.Android/Adapters/SqLiteTableAdapter.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

using Android.App;
using Android.Content;
using Android.OS;
using Android.Runtime;
using Android.Views;
using Android.Widget;
using PerfTest2Xamarin.Enums;
using PerfTest2Xamarin.Utilities;

namespace PerfTest2Xamarin.Adapters
{
    public class SqLiteTableAdapter : BaseAdapter<string>
    {

        private Context context;
        private string directory;
        private IList<string> records;
        private SqLiteDisplayType displayType;

        public SqLiteTableAdapter(Context context, string directory, SqLiteDisplayType displayType)
        {
            this.context = context;
            this.directory = directory;
            this.displayType = displayType;
            LoadRecords();
        }

        private void LoadRecords()
        {
            SqLiteUtilitiesAlt utilities;

            utilities = new SqLiteUtilitiesAlt(this.context);
            try
            {
                utilities.OpenConnection();
                if (displayType == SqLiteDisplayType.ShowAll)
                {
                    records = utilities.GetAllRecords();
                }
                else
                {
                    records = utilities.GetRecordsWith1();
                }
                utilities.CloseConnection();
            }
            catch (Exception ex)
            {
                // Bah, it's not production code... 
            }
        }


        public override string this[int position]
        {
            get { return records[position]; }
        }

        public override int Count
        {
            get { return records.Count; }
        }

        public override long GetItemId(int position)
        {
            return position;
        }

        public override View GetView(int position, View convertView, ViewGroup parent)
        {
            if (convertView == null)
            {
                LayoutInflater inflater = (LayoutInflater)context.GetSystemService(Context.LayoutInflaterService);
                convertView = inflater.Inflate(Android.Resource.Layout.SimpleListItem1, null);
            }

            TextView txtItem = (TextView)convertView.FindViewById(Android.Resource.Id.Text1);
            txtItem.Text = records[position];

            return convertView;
        }
    }
}